In [1]:
from utils import load_finetuned_qwen, round_down_to_2_decimals
from title_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
def print_model_scores(model):
    scores = get_scores(model)
    print(f"MRR: {round_down_to_2_decimals(scores["perfect_match_score"]):.2f}")
    print(f"Precision@5: {round_down_to_2_decimals(scores["precision_at_5"]):.2f}")

Base scibert

In [3]:
print_model_scores("allenai/scibert_scivocab_uncased")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 17933.23it/s]
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 

Notes:
- UNEXPECTED:	can be ignored when loading from different task/architecture; not ok if you expect identical arch.


MRR: 0.01
Precision@5: 0.05


Vanilla scibert

In [4]:
print_model_scores("models/vanilla-scibert")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 2793.09it/s]


MRR: 0.05
Precision@5: 0.25


Jargon-layman scibert

In [5]:
print_model_scores("models/jargon-layman-scibert-1.00")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 2375.94it/s]


MRR: 0.06
Precision@5: 0.26


Vanilla qwen

In [6]:
print_model_scores(
    load_finetuned_qwen("models/vanilla-qwen")
)

Loading weights:   0%|          | 1/310 [00:00<01:05,  4.69it/s]/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/bitsandbytes/backends/cuda/ops.py:213: FutureWarning: _check_is_size will be removed in a future PyTorch release along with guard_size_oblivious.     Use _check(i >= 0) instead.
  torch._check_is_size(blocksize)
Loading weights: 100%|██████████| 310/310 [00:01<00:00, 263.68it/s]


MRR: 0.04
Precision@5: 0.17


Jargon-layman qwen

In [7]:
print_model_scores(
    load_finetuned_qwen("models/jargon-layman-qwen")
)

Loading weights:   0%|          | 0/310 [00:00<?, ?it/s]/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/bitsandbytes/backends/cuda/ops.py:213: FutureWarning: _check_is_size will be removed in a future PyTorch release along with guard_size_oblivious.     Use _check(i >= 0) instead.
  torch._check_is_size(blocksize)
Loading weights: 100%|██████████| 310/310 [00:00<00:00, 1047.67it/s]


MRR: 0.04
Precision@5: 0.17


# Results

| Model | MRR | Precision@5 |
|---|---|---|
| allenai/scibert_scivocab_uncased | 0.01 | 0.05 |
| vanilla-scibert | 0.05 | 0.25 |
| jargon-layman-scibert | 0.06 | 0.26 |
| vanilla-qwen | 0.04 | 0.17 |
| jargon-layman-qwen | 0.04 | 0.17 |